turborepo-tests/integration/tests/run/gitignored-inputs.t

```diff
@@ -0,0 +1,39 @@
+Setup
+  $ . ${TESTDIR}/../../../helpers/setup.sh
+  $ . ${TESTDIR}/../_helpers/setup_monorepo.sh $(pwd)
+
+Use our custom turbo config which has foo.txt as an input to the build command
+  $ . ${TESTDIR}/../../../helpers/replace_turbo_config.sh $(pwd) "gitignored-inputs.json"
+
+Create a internal.txt for the util package and add it to gitignore
+This fiel is already part of our turbo config.
+  $ echo "hello world" >> packages/util/internal.txt
+  $ echo "packages/util/internal.txt" >> ${PWD}/.gitignore
+  $ if [[ "$OSTYPE" == "msys" ]]; then dos2unix --quiet packages/util/internal.txt; fi
+  $ git add . && git commit --quiet -m  "add internal.txt"
+
+Some helper functions to parse the summary file
+  $ source "$TESTDIR/../_helpers/run-summary-utils.sh"
+
+Just run the util package, it's simpler
+  $ ${TURBO} run build --filter=util --output-logs=hash-only --summarize | grep "util:build: cache"
+  util:build: cache miss, executing 2f6ab59379ddcb93
+
+  $ FIRST=$(/bin/ls .turbo/runs/*.json | head -n1)
+  $ echo $(getSummaryTaskId $FIRST "util#build") | jq -r '.inputs."internal.txt"'
+  3b18e512dba79e4c8300dd08aeb37f8e728b8dad
+
+Cleanup the runs folder so we don't have to select the correct file for the second run
+  $ rm -rf .turbo/runs
+
+Change the content of internal.txt
+  $ echo "changed!" >> packages/util/internal.txt
+
+Hash does not change, because it is gitignored
+  $ ${TURBO} run build --filter=util --output-logs=hash-only --summarize | grep "util:build: cache"
```

Comment:
The hash should been changed whatever is it included in the gitignored

Reply:
If you want this behaviour our recommended method is to specify the inputs for the task yourself. By default we do not include ignored files because it is very unlikely that you will get cache hits in CI for example.

Reply:
@Cow258 if you see the next line after this one, it is:
```
util:build: cache miss, executing a26c95f27f26f89c
```

and if you see the line before I changed the `internal.txt` file on line 20 of this file, it is:
```
  util:build: cache miss, executing 2f6ab59379ddcb93
```

so the hash _did_ change